### PR TITLE
Add support for alternative operators 'and, or, not' (#94)

### PR DESCRIPTION
**Pull request by azabiong**

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1146,6 +1148,25 @@ module.exports = grammar(C, {
       $.literal_suffix
     ),
 
+    alternative_unary: $ => prec.left(PREC.UNARY, seq(
+      field('operator', 'not'),
+      field('argument', $._expression)
+    )),
+
+    alternative_binary: $ => {
```
Comment (mliszcz) [inline grammar.js:1156]: How about adding those alternative operators to unary_expression / binary_expression rules inherited from the C grammar? I guess having the same parse tree could simplify some queries.

Perhaps we could even add them directly to the C grammar? C95 has these defined in a header file as preprocessor macros, see: https://en.wikipedia.org/wiki/C_alternative_tokens

File: grammar.js [hunk 2]
```diff
@@ -1146,6 +1148,25 @@ module.exports = grammar(C, {
...
+      const table = [
+        ['or',  PREC.LOGICAL_OR],
```
Comment (mliszcz) [inline grammar.js:1158]: Please consider adding all of them. This should not make the PR any more complex and we will have full support for alternative operators.

**Review by mliszcz — COMMENTED** [review]
I took the liberty to leave some comments in the hope of speeding-up the review process.
Please take them with a grain of salt since I am not affiliated with this project and the final word belongs to the maintainers.

**Comment by mliszcz** [comment]
Hi @jdrouhard

I see you are assigned as a reviewer, could you please clarify the following:
1. are alternative operator names even considered for being accepted intro tree-sitter-cpp?
2. what can we do to accelerate the review/merge process? I'm using @azabiong's patch (with neovim) for more than two months now without any issues.

Sorry for pinging and disturbing both of you, but the PR got stalled and for me it is an important one.
Thanks!

**Comment by jdrouhard** [comment]
I went ahead and did an alternate implementation of this in #179.

The alternative operators shouldn't introduce new tree nodes for binary and unary expressions. I also added more of the alternate operators and made them work with operator calls/overloads and assignment expressions.

**Comment by azabiong** [comment]
Great! Thanks

**Pull request closed without merging**